[Clang][Doc][NFC] Improve `-offload-compress` documentation and error message (#17990)
intel/llvm: sycl/doc/GetStartedGuide.md

Both kernel fusion and JIT compilation of AMD and Nvidia kernels are currently
not yet supported on the Windows platform.

### Build DPC++ toolchain with device image compression support

> Device image compression enables the compression of device code (SYCL Kernels) during compilation and decompressing them on-demand at application runtime.

Reviewer: I would reword it a little:
```suggestion
Device image compression enables the compression of device code (SYCL Kernels) during compilation and decompressing them on-demand during the execution of the corresponding SYCL application.
```

This reduces the size of fat binaries for both Just-in-Time (JIT) and Ahead-of-Time (AOT) compilation. Refer to the [blog post](https://www.intel.com/content/www/us/en/developer/articles/technical/sycl-compilation-device-image-compression.html) for more details on this feature.

To enable device image compression, you need to build the DPC++ toolchain with the
zstd compression library. By default, zstd is optional for DPC++ builds i.e. CMake will search for zstd installation but if not found, it will not fail the build
and this feature will simply be disabled.

> To override this behavior and force the build to use zstd, you can use the `--use-zstd` flag in the `configure.py` script.

Reviewer: ```suggestion
To override this behavior and force the build to use zstd, you can use the `--use-zstd` flag in the `configure.py` script or by adding `-DLLVM_ENABLE_ZSTD=FORCE_ON` to the CMake configuration command.
```

#### How to obtain zstd?

Minimum zstd version that we have tested with is *1.4.8*.

**Linux**:

You can install zstd using the package manager of your distribution. For example, on Ubuntu, you can run:
```sh
sudo apt-get install libzstd-dev
```

> Note that the libzstd-dev package provided on Ubuntu 24.04 has a bug and the zstd static library is not built with `-fPIC` flag. Linking to this library will result in a build failure. For example: [Issue#15935](https://github.com/intel/llvm/issues/15935). As an alternative, zstd can be built from source either manually or by using the [build_zstd_1_5_6_ub24.sh](https://github.com/intel/llvm/blob/sycl/devops/scripts/build_zstd_1_5_6_ub24.sh) script.

Reviewer 2: ```suggestion
Note that the libzstd-dev package provided on Ubuntu 24.04 has a bug and the zstd static library is not built with the `-fPIC` flag. Linking to this library will result in a build failure. For example: [Issue#15935](https://github.com/intel/llvm/issues/15935). As an alternative, zstd can be built from source either manually or by using the [build_zstd_1_5_6_ub24.sh](https://github.com/intel/llvm/blob/sycl/devops/scripts/build_zstd_1_5_6_ub24.sh) script.
```

> Note that the libzstd-dev package provided on Ubuntu 24.04 has a bug and the zstd static library is not built with `-fPIC` flag. Linking to this library will result in a build failure. For example: [Issue#15935](https://github.com/intel/llvm/issues/15935). As an alternative, zstd can be built from source either manually or by using the [build_zstd_1_5_6_ub24.sh](https://github.com/intel/llvm/blob/sycl/devops/scripts/build_zstd_1_5_6_ub24.sh) script.

Reviewer 2: might be helpful to link to the launchpad bug here so people can know when its fixed

**Windows**

> For Windows, prebuilt zstd binaries can be obtained from the [facebook/zstd](https://github.com/facebook/zstd/releases/tag/v1.5.6) release page. After obtaining the zstd binaries, you can add
> to the `PATH` environment variable the path to the zstd installation directory.

Reviewer: ```suggestion
For Windows, prebuilt zstd binaries can be obtained from the [facebook/zstd](https://github.com/facebook/zstd/releases/tag/v1.5.6) release page. After obtaining the zstd binaries, you can add the path to the zstd installation directory to the `PATH` environment variable.
```
